{"ast":null,"code":"var _jsxFileName = \"/home/noor/study/crwn-clothing/client/src/components/error-boundary/error-boundary.component.jsx\";\nimport React from \"react\";\nimport { jsxDEV as _jsxDEV } from \"react/jsx-dev-runtime\";\n\nclass ErrorBoundary extends React.Component {\n  constructor() {\n    super();\n    this.state = {\n      hasErrored: false\n    };\n  }\n\n  static getDerivedStatFromError(error) {\n    return {\n      hasErrored: true\n    };\n  }\n\n  componentDidCatch(error, info) {\n    console.log(error);\n  }\n\n  render() {\n    if (this.state.hasErrored) {\n      console.log(\"\");\n      return /*#__PURE__*/_jsxDEV(\"div\", {\n        children: \" Something went wrong\"\n      }, void 0, false, {\n        fileName: _jsxFileName,\n        lineNumber: 24,\n        columnNumber: 18\n      }, this);\n    }\n\n    return this.props.children;\n  }\n\n}\n\nexport default ErrorBoundary;","map":{"version":3,"sources":["/home/noor/study/crwn-clothing/client/src/components/error-boundary/error-boundary.component.jsx"],"names":["React","ErrorBoundary","Component","constructor","state","hasErrored","getDerivedStatFromError","error","componentDidCatch","info","console","log","render","props","children"],"mappings":";AAAA,OAAOA,KAAP,MAAkB,OAAlB;;;AAGA,MAAMC,aAAN,SAA4BD,KAAK,CAACE,SAAlC,CAA4C;AACxCC,EAAAA,WAAW,GAAG;AACV;AAEA,SAAKC,KAAL,GAAa;AACTC,MAAAA,UAAU,EAAG;AADJ,KAAb;AAGH;;AAE6B,SAAvBC,uBAAuB,CAACC,KAAD,EAAQ;AAClC,WAAO;AAAEF,MAAAA,UAAU,EAAG;AAAf,KAAP;AACH;;AAEDG,EAAAA,iBAAiB,CAAED,KAAF,EAASE,IAAT,EAAgB;AAC7BC,IAAAA,OAAO,CAACC,GAAR,CAAYJ,KAAZ;AACH;;AAEDK,EAAAA,MAAM,GAAG;AACL,QAAG,KAAKR,KAAL,CAAWC,UAAd,EAAyB;AACrBK,MAAAA,OAAO,CAACC,GAAR,CAAY,EAAZ;AACF,0BAAO;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,cAAP;AACD;;AAED,WAAO,KAAKE,KAAL,CAAWC,QAAlB;AACH;;AAxBuC;;AA2B5C,eAAeb,aAAf","sourcesContent":["import React from \"react\";\n\n\nclass ErrorBoundary extends React.Component {\n    constructor() {\n        super();\n\n        this.state = {\n            hasErrored : false,\n        }\n    }\n\n    static getDerivedStatFromError(error) {\n        return { hasErrored : true}\n    }\n\n    componentDidCatch( error, info ) {\n        console.log(error);\n    }\n\n    render() {\n        if(this.state.hasErrored){\n            console.log(\"\");\n          return <div> Something went wrong</div>\n        }\n\n        return this.props.children;\n    }\n}\n\nexport default ErrorBoundary;"]},"metadata":{},"sourceType":"module"}